blip-tap.md

```diff
@@ -919,149 +919,307 @@ order to clamp down on price volatility exposure.
 TODO(roasbeef): explain AMP/keyspend, basically unsolicited quote req, can
 accept if favorable
 
-
 #### RFQ Negotiation
 
-##### Request For Quote (`tap_rfq`)
+This section describes the Request-For-Quote (RFQ) negotiation process. A
+successful outcome of the RFQ process between two peers is an agreed currency
+exchange rate. This rate can be used to convert a TAP asset amount into BTC, or
+vice versa.
 
-When a receiver wishes to receive `N` units of TAP asset ID `asset_id`, a new
-p2p message `tap_rfq` is sent with the following structure:
+##### Request For Quote (`tap_rfq`)
 
-1. type: ?? (`tap_rfq`)
-2. data:
-     * [`32*byte`:`rfq_id`]
-     * [`32*byte`:`asset_id`]
-     * [`BigSize`:`asset_amt`]
-     * [`BigSize`:`suggested_rate_tick`]
+An RFQ negotiation is initiated when a requesting peer sends a new quote request
+p2p message called `tap_rfq` to their channel counterparty.
+
+In this message, the peer includes the asset they wish to obtain (identified by
+`in_asset_id`/`in_asset_group_key`) and the asset they wish to divest
+(identified by `out_asset_id`/`out_asset_group_key`).
+
+The `tap_rfq` message has a BOLT 01 wire message type integer of 52884, falling
+within the custom message range. Its TLV stream consists of the following TLV
+fields:
+
+| TLV Type | Field Name                | Field Type |
+|----------|---------------------------|------------|
+| 0        | version                   | 1*byte     |
+| 2        | rfq_id                    | 32*byte    |
+| 4        | transfer_type             | 1*byte     |
+| 6        | expiry                    | 8*byte     |
+| 9        | in_asset_id               | 32*byte    |
+| 11       | in_asset_group_key        | 33*byte    |
+| 13       | out_asset_id              | 32*byte    |
+| 15       | out_asset_group_key       | 33*byte    |
+| 16       | max_transferable_in_asset | BigSize    |
```

Comment:
As it stands, I don't believe we require or use a minimum. I'm not entirely clear on the value of having one, especially since the quote might not be used at all. A minimum volume doesn't seem to provide any protection for the edge node, and it appears to complicate MPP payments unnecessarily.

Could you please clarify the use case you had in mind?

Reply:
> Could you please clarify the use case you had in mind?

IIRC even with the RFQ revamp, given a set of params (BTC price, asset price) it's possible that 1 asset unit cannot be sent over the LN. This happens when 1 unit <<< 1 mSAT. Therefore, we want to be able to prevent such a send attempt in the first place. Adding a min sendable value here can help us handle such scenarios (send attempt and rfq negotiation rejected).